Tests/Core/AWSClientRuntimeTests/Auth/CredentialsProvidersTests/SSOCredentialsProviderTests.swift

```diff
@@ -0,0 +1,37 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import ClientRuntime
+import Foundation
+import XCTest
+
+@_spi(FileBasedConfig) @testable import AWSClientRuntime
+
+class SSOCredentialsProviderTests: XCTestCase {
+    let configPath = Bundle.module.path(forResource: "sso_tests", ofType: nil)!
+    let credentialsPath = Bundle.module.path(forResource: "credentials", ofType: nil)!
+    
+    func testCreateCredentialsProviderSSOLegacyProfile() async throws {
+        let provider = try SSOCredentialsProvider(
+            profileName: "user",
+            configFilePath: configPath,
+            credentialsFilePath: credentialsPath)
+        XCTAssertNotNil(provider)
```

Comment:
nit: line 22 and same in the test below closing ) should be on new line

Just curious, why don't we call provider.getCredentials() anymore? I was thinking the test is

```
let provider = try SSOCredentialsProvider(
            profileName: "user",
            configFilePath: configPath,
            credentialsFilePath: credentialsPath
)
let credentials = try provider.getCredentials()

XCTAssert(credentials.field, expectedValue)
...
```

Reply:
I think the unit tests here should also 'mock' (swift equivalent) responses from getCredentials() and then make sure that our credentials are loaded based on that response giving us something valid or throwing an error. Right now the scope seems too small to me

Reply:
After discussion on slack I have better idea on what to do now - will be making changes and pushing them.

Reply:
For this provider, since getCredentials() is a method defined by the CRT which makes a service call, we are limited in our ability to sufficiently unit test that logic in the SDK. For test coverage here I am fine with unit tests covering instantiation alone of SSOCredentialsProvider given that we plan to create integration tests in a fast follow up.

Reply:
Instead of creating provider and checking that its not nil, can either omit that or ideally confirm that all members on the created object are set properly